ibis/pandas/execution.py

```diff
@@ -0,0 +1,438 @@
+
+
+_JOIN_TYPES = {
+    ops.LeftJoin: 'left',
+    ops.InnerJoin: 'inner',
+    ops.OuterJoin: 'outer',
+}
+
+
+@execute_node.register(ops.Join, pd.DataFrame, pd.DataFrame)
+def execute_join(op, left, right, scope=None):
+    try:
+        how = _JOIN_TYPES[type(op)]
+    except KeyError:
+        raise NotImplementedError('{} not supported'.format(type(op).__name__))
+
+    left_on = []
+    right_on = []
+
+    for predicate in op.predicates:
+        pred_op = predicate.op()
+        if not isinstance(pred_op, ops.Equals):
+            raise TypeError(
+                'Only equality join predicates supported with pandas'
+            )
+        left_key = pred_op.left._name
+        right_key = pred_op.right._name
+        left_on.append(left_key)
+        right_on.append(right_key)
+
+    return pd.merge(left, right, how=how, left_on=left_on, right_on=right_on)
+
+
+_BINARY_OPERATIONS = {
+    ops.Greater: operator.gt,
+    ops.Less: operator.lt,
+    ops.LessEqual: operator.le,
+    ops.GreaterEqual: operator.ge,
+    ops.Equals: operator.eq,
+    ops.NotEquals: operator.ne,
+
+    ops.And: operator.and_,
+    ops.Or: operator.or_,
+    ops.Xor: operator.xor,
+
+    ops.Add: operator.add,
+    ops.Subtract: operator.sub,
+    ops.Multiply: operator.mul,
+    ops.Divide: operator.truediv,
+    ops.FloorDivide: operator.floordiv,
+    ops.Modulus: operator.mod,
+    ops.Power: operator.pow,
+}
+
+
+@execute_node.register(
+    ops.BinaryOp, pd.Series, (pd.Series, numbers.Real) + six.string_types
+)
+def execute_binary_operation_series(op, left, right, scope=None):
+    return _BINARY_OPERATIONS[type(op)](left, right)
+
+
+@execute_node.register(ops.Not, pd.Series)
+def execute_not_series(op, data, scope=None):
+    return ~data
+
+
+@execute_node.register(ops.Strftime, pd.Timestamp, six.string_types)
+def execute_strftime_timestamp_str(op, data, format_string, scope=None):
+    return data.strftime(format_string)
+
+
+@execute_node.register(ops.Strftime, pd.Series, six.string_types)
+def execute_strftime_series_str(op, data, format_string, scope=None):
+    return data.dt.strftime(format_string)
+
+
+@execute_node.register(ops.ExtractTimestampField, pd.Timestamp)
+def execute_extract_timestamp_field_timestamp(op, data, scope=None):
+    field_name = type(op).__name__.lower().replace('extract', '')
+    return getattr(data, field_name)
+
+
+@execute_node.register(ops.ExtractMillisecond, pd.Timestamp)
+def execute_extract_millisecond_timestamp(op, data, scope=None):
+    return int(data.microsecond // 1000.0)
+
+
+@execute_node.register(ops.ExtractTimestampField, pd.Series)
+def execute_extract_timestamp_field_series(op, data, scope=None):
+    field_name = type(op).__name__.lower().replace('extract', '')
+    return getattr(data.dt, field_name)
+
+
+@execute_node.register(ops.NullIfZero, pd.Series)
+def execute_null_if_zero_series(op, data, scope=None):
+    result = data.copy()
+    result[result == 0] = np.nan
+    return result
```

Comment:
Probably not. This happened to be the first thing that came to mind. My pandas-fu is a little rusty.